pkg/resources/openstack/floatingip.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	l3floatingip "github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+	"k8s.io/kops/pkg/resources"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/openstack"
+)
+
+const (
+	typeFloatingIP = "FloatingIP"
+)
+
+func DeleteFloatingIP(cloud fi.Cloud, r *resources.Resource) error {
+	return cloud.(openstack.OpenstackCloud).DeleteFloatingIP(r.ID)
+}
+
+func DeleteL3FloatingIP(cloud fi.Cloud, r *resources.Resource) error {
+	return cloud.(openstack.OpenstackCloud).DeleteL3FloatingIP(r.ID)
+}
+
+func (os *clusterDiscoveryOS) listL3FloatingIPs(routerID string) ([]*resources.Resource, error) {
+	var resourceTrackers []*resources.Resource
```

Comment:
We'll have to be careful here if we allow binding to an existing IP.  I'm guessing that is not currently supported though?

Reply:
binding to existing ip addresses are not supported (yet). IF that would be possible, then we should not delete floatingips